pkg/sql/mvcc_backfiller.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/rowenc"
+	"github.com/cockroachdb/errors"
+)
+
+// Merge merges the entries in the provide span sourceSpan from the index with
+// sourceID into the index with destinationID.
+func (sc *SchemaChanger) Merge(
+	ctx context.Context,
+	codec keys.SQLCodec,
+	table catalog.TableDescriptor,
+	sourceID descpb.IndexID,
+	destinationID descpb.IndexID,
+	sourceSpan roachpb.Span,
+) error {
+	sourcePrefix := rowenc.MakeIndexKeyPrefix(codec, table, sourceID)
+	prefixLen := len(sourcePrefix)
+	destPrefix := rowenc.MakeIndexKeyPrefix(codec, table, destinationID)
+
+	const pageSize = 1000
```

Comment:
I wonder if we want to make this a cluster setting tunable at some point.  No need to do that in this PR as I imagine the exact details will change as you work on the checkpointing and distribution bits.